### PR TITLE
in ICG: Add FileSkipped preprocessor callback to FindTrickICG

### PR DESCRIPTION
**Pull request by spfennell**
See issue #608

### CODE REVIEW

File: trick_source/codegen/Interface_Code_Gen/FindTrickICG.cpp [hunk 1]
```diff
@@ -35,6 +35,29 @@ void FindTrickICG::FileChanged(clang::SourceLocation Loc, FileChangeReason Reaso
...
+#else
+    std::string file_name = SkippedFile.getName().str() ;
+#endif
+    std::string file_path(almostRealPath(file_name.c_str()));
+    // Check if skipped header is in Compat15
+    if(hsd.isPathInCompat15(file_path)) {
+        // for each header in the stack, mark them as being exposed to TRICK_ICG
+        std::vector<std::string>::iterator it ;
+        for ( it = included_files.begin() ; it != included_files.end() ; it++ ) {
+            hsd.addTrickICGFoundFile(*it) ;
```
Comment (dbankieris) [inline trick_source/codegen/Interface_Code_Gen/FindTrickICG.cpp:55]: Embrace C++11's foreach syntax.

```suggestion
        for (std::string& file : included_files) {
            hsd.addTrickICGFoundFile(file);
```

Comment (spfennell) [reply trick_source/codegen/Interface_Code_Gen/FindTrickICG.cpp]: Done

**Comment by spfennell** [comment]
I'll merge this after I write a test for it

**Comment by spfennell** [comment]
Tests fail as expected on test-only branch [608-add-test-ci-should-fail](https://github.com/nasa/trick/tree/608-add-test-ci-should-fail).

**Review by dbankieris — APPROVED** [review]
Love the test sim!

**Comment by spfennell** [comment]
closes #608

**Pull request merged**